charms/istio-pilot/src/charm.py

```diff
@@ -380,12 +386,18 @@ def _get_interfaces(self):
             raise ErrorWithStatus(err, BlockedStatus)
         return interfaces
 
-    def _get_ingress_auth_data(self) -> dict:
+    def _get_ingress_auth_data(self, event) -> dict:
         """Retrieve the ingress-auth relation data without touching other interface data.
 
         This is a workaround to ensure that errors in other relation data, such as an incomplete
         ingress relation, do not block us from retrieving the ingress-auth data.
         """
+        # Do not process data if this is a relation-broken event for this relation
+        if (
+            isinstance(event, RelationBrokenEvent)
```

Comment:
Why not use observe and react to the relation broken event?

Reply:
I think I would agree with @DnPlas 

we can create another observer and put also this logic from `_reconcile_ingress_auth` there:
```python
        if len(ingress_auth_data) == 0:
            self.log.info("No ingress-auth data found - deleting any existing EnvoyFilter")
            _remove_envoyfilter(envoyfilter_name)
            _remove_envoyfilter(name=envoyfilter_name, namespace=self.model.name)
            return
```

Reply:
or no, since independent if we have ingress data or not we proceed to 
call  `_reconcile_gateway`

Reply:
We do observe the [`ingress-auth` relation_broken event](https://github.com/canonical/istio-operators/blob/4994fbf29bff048195e48cc55ee64995eebcbf16/charms/istio-pilot/src/charm.py#L144).  

When processing any relation_broken event, the relation data bag for that departing application *still exists and contains data*, so we cannot handle the event the same as other ingress-auth events.  If we don't treat this case special, it will still create the EnvoyFilter from the ingress-auth data even though the user is asking to turn that off.  

This is similar to [what we do for the `ingress` relation](https://github.com/canonical/istio-operators/blob/4994fbf29bff048195e48cc55ee64995eebcbf16/charms/istio-pilot/src/charm.py#L451), where we pop the breaking relation's data out of the data bag before processing.  But in the case of ingress-auth since it only supports being related to a single app, we can just return an empty dict instead.